core/test/functional/routes/api/posts_spec.js

```diff
@@ -1321,25 +1321,78 @@ describe('Post API', function () {
                 });
         });
 
+        describe('Add', function () {
+            it('can add own post', function (done) {
+                var post = {
+                    title: 'Freshly added',
+                    slug: 'freshly-added',
+                    author_id: author.id
+                };
+
+                request.post(testUtils.API.getApiQuery('posts/'))
+                    .set('Authorization', 'Bearer ' + authorAccessToken)
+                    .send({posts: [post]})
+                    .expect('Content-Type', /json/)
+                    .expect('Cache-Control', testUtils.cacheRules.private)
+                    .expect(201)
+                    .end(function (err, res) {
+                        if (err) {
+                            return done(err);
+                        }
+
+                        var postBody = res.body;
+
+                        res.headers['x-cache-invalidate'].should.eql('/p/' + postBody.posts[0].uuid + '/');
+                        should.exist(postBody);
+
+                        testUtils.API.checkResponse(postBody.posts[0], 'post');
+                        done();
+                    });
+            });
+
+            it('CANNOT add post with other author ID', function (done) {
+                var post = {
+                    title: 'Freshly added',
+                    slug: 'freshly-added',
+                    author_id: 1
+                };
+
+                request.post(testUtils.API.getApiQuery('posts/'))
+                    .set('Authorization', 'Bearer ' + authorAccessToken)
+                    .send({posts: [post]})
+                    .expect('Content-Type', /json/)
+                    .expect('Cache-Control', testUtils.cacheRules.private)
+                    .expect(403)
+                    .end(function (err, res) {
+                        if (err) {
+                            return done(err);
+                        }
+
+                        should.exist(res.body.errors);
+                        res.body.errors[0].errorType.should.eql('NoPermissionError');
```

Comment:
Should we return a more specific error message?
Because the message says if i modify the author: "You do not have permission to edit posts".

Reply:
Hmm or maybe not. Because i am not able to change my author id in the admin panel, there is no UI if i am an author.

And if somebody else changes my id, the message is correct.

Reply:
Ideally, yes, it should be more specific. In practice, I can't see a way to do this that doesn't require more refactoring & I don't think it's important enough given that there is no UI.

Something for future TODOs - have added a note to #9043

Reply:
Agree 👍